### PR TITLE
Add Clarifai's types as new types in airtable: issue #214

### PR DESCRIPTION
**Pull request by orjwan98**
References issue #214 

What I did in this PR:
**client/src/components/GetDetails/index.js**
- check if item's typeId is empty or not. 
if yes:
 - save it in airtable as a new type and set its newfound Id in the state. 
 - save it to airtable (still not tested).

if not: 
- save item normally as you would do. 

------------------------------------------
**server/router.js**

- added `/POST` route and hander to router. 

-------------------------------------------
**server/controllers/addType.js**

- Handler for the `/POST` endpoint. 
- recieves shortcut and name. 
- adds a new type to airtable where name = itemType and shortcut = "New Type"
- Retrieves the Id from the new added record and sends it to FE.

### CODE REVIEW

**Review by amanshawar — APPROVED** [review]

File: client/src/components/GetDetails/index.js [hunk 1]
```diff
@@ -114,6 +114,7 @@ class GetDetails extends Component {
       const price = this.state.selected_price.concat(this.state.selected_currency);
       const inputs = {
         type: this.state.selected_itemType,
+        typeId: this.state.selected_itemType.id,
```
Comment (Hadeel-Salamin) [inline client/src/components/GetDetails/index.js]: /add-item endpoint only receive "type" in request so u have to set this.state.selected_itemType.id to type, remove typeId

File: client/src/components/GetDetails/index.js [hunk 2]
```diff
@@ -129,7 +130,22 @@ class GetDetails extends Component {
         pattern: this.state.selected_patterns,
       };
 
-      if (cookie) {
+      if (inputs.typeId === '') {
+        axios.post('/add-type', { name: inputs.type.name, shortcut: 'New Type' }).then((res) => {
```
Comment (Hadeel-Salamin) [inline client/src/components/GetDetails/index.js]: oh I see you are using type here but still how u handle adding the type id in back end while you don't have this value in the request

Comment (Hadeel-Salamin) [reply client/src/components/GetDetails/index.js]: if you did not get what I mean, check this endpoint on BE, match FE with BE please

File: client/src/components/GetDetails/index.js [hunk 3]
```diff
@@ -129,7 +130,22 @@ class GetDetails extends Component {
...
+          const { typeId } = res.data;
+          this.setState({ typeId }, () => {
+            if (cookie) {
+              axios.post('/add-item', inputs).then(({ data }) => {
```
Comment (Hadeel-Salamin) [inline client/src/components/GetDetails/index.js]: if you did not get what I mean, check this endpoint on BE, match FE with BE please

Comment (orjwan98) [reply client/src/components/GetDetails/index.js]: Yeah will get around to do it in the next pull request, don't worry!

File: client/src/components/GetDetails/index.js (hunk 1 above)
Comment (orjwan98) [reply client/src/components/GetDetails/index.js]: Okay I modified it in both cases. Take a look now.

File: client/src/components/GetDetails/index.js [hunk 4]
```diff
@@ -129,7 +129,22 @@ class GetDetails extends Component {
         pattern: this.state.selected_patterns,
       };
 
-      if (cookie) {
+      if (inputs.type === '') {
+        axios.post('/add-type', { name: this.state.selected_itemType, shortcut: 'New Type' }).then((res) => {
```
Comment (Hadeel-Salamin) [inline client/src/components/GetDetails/index.js]: you meant .name right?

File: client/src/components/GetDetails/index.js [hunk 5]
```diff
@@ -129,7 +129,22 @@ class GetDetails extends Component {
...
+          const { typeId } = res.data;
+          this.setState({ typeId }, () => {
```
Comment (Hadeel-Salamin) [inline client/src/components/GetDetails/index.js]: I can't see how you updating this in state and send it later by inputs to BE?

File: client/src/components/GetDetails/index.js [hunk 6]
```diff
@@ -129,7 +129,23 @@ class GetDetails extends Component {
         pattern: this.state.selected_patterns,
       };
 
-      if (cookie) {
+      if (inputs.type === '') {
+        axios.post('/add-type', { name: this.state.selected_itemType.name, shortcut: 'New Type' }).then((res) => {
+          const { typeId } = res.data;
+          this.setState({ this.state.selected_itemType.id:`${typeId}` }, () => {
```
Comment (Hadeel-Salamin) [inline client/src/components/GetDetails/index.js]: still the id is not updated in inputs

File: client/src/components/GetDetails/index.js [hunk 7]
```diff
@@ -129,7 +129,23 @@ class GetDetails extends Component {
...
+            console.log(this.state.selected_itemType.id)
```
Comment (Hadeel-Salamin) [inline client/src/components/GetDetails/index.js]: remove console.log

**Review by amanshawar — APPROVED** [review]

**Review by jema28 — APPROVED** [review]

**Pull request merged**